conn.go

```diff
@@ -113,6 +117,9 @@ func (conn *Connection) Leader() (string, error) {
 	if conn.hasBeenClosed {
 		return "", errClosed
 	}
+	if conn.disableClusterDiscovery {
```

Comment:
Why this change? What is your thinking?

Reply:
This is so that updateClusterInfo isn't called, because if it is, we're back to the same problem. I don't mind if it happens since we're not using this function, but it feels wrong to have the behavior change just because this function got used. Same with Peers.